packages/shared/src/components/auth/SocialRegistrationForm.tsx

```diff
@@ -60,11 +64,40 @@ export const SocialRegistrationForm = ({
   isLoading,
 }: SocialRegistrationFormProps): ReactElement => {
   const { trackEvent } = useContext(AnalyticsContext);
-  const { user } = useContext(AuthContext);
+  const { user, updateUser } = useContext(AuthContext);
   const [nameHint, setNameHint] = useState<string>(null);
   const [usernameHint, setUsernameHint] = useState<string>(null);
   const [twitterHint, setTwitterHint] = useState<string>(null);
+  const [name, setName] = useState<string>('');
   const isAuthorOnboarding = trigger === AuthTriggers.Author;
+  const { requestMethod } = useRequestProtocol();
+  const usernameQueryKey = ['generateUsername', name];
```

Comment:
This code is now pretty much duplicate, except the success handler being different. 
I'd suggest abstracting it to a custom hook so we can re-use it.

Maybe make it accept a callback function to handle both cases.